core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c

```diff
@@ -940,32 +947,13 @@ static bool __clk_is_enabled(struct stm32mp1_clk_gate const *gate)
 
 static bool clock_is_always_on(unsigned long id)
 {
-	COMPILE_TIME_ASSERT(CK_HSE == 0 &&
-			    (CK_HSE + 1) == CK_CSI &&
-			    (CK_HSE + 2) == CK_LSI &&
-			    (CK_HSE + 3) == CK_LSE &&
-			    (CK_HSE + 4) == CK_HSI &&
-			    (CK_HSE + 5) == CK_HSE_DIV2 &&
-			    (PLL1_P + 1) == PLL1_Q &&
-			    (PLL1_P + 2) == PLL1_R &&
-			    (PLL1_P + 3) == PLL2_P &&
-			    (PLL1_P + 4) == PLL2_Q &&
-			    (PLL1_P + 5) == PLL2_R &&
-			    (PLL1_P + 6) == PLL3_P &&
-			    (PLL1_P + 7) == PLL3_Q &&
-			    (PLL1_P + 8) == PLL3_R);
-
-	if (id <= CK_HSE_DIV2 || (id >= PLL1_P && id <= PLL3_R))
-		return true;
+	size_t n = 0;
 
-	switch (id) {
-	case CK_AXI:
-	case CK_MPU:
-	case CK_MCU:
-		return true;
-	default:
-		return false;
-	}
+	for (n = 0; n < ARRAY_SIZE(stm32mp1_clk_on); n++)
+		if (stm32mp1_clk_on[n] == id)
+			return true;
+
+	return false;
 }
```

Comment:
Why `DECLARE_KEEP_PAGER(stm32mp1_clk_on);` and not `DECLARE_KEEP_PAGER(clock_is_always_on);`?
Could `stm32mp1_clk_on` be local to `clock_is_always_on`?

Reply:
Right, i was to hurry on this. No need for `DECLARE_KEEP_PAGER(stm32mp1_clk_on)`, same for   `clock_is_always_on()`: they will already be embedded into unpaged sections if needed using the standard dependency chains.